### PR TITLE
add route for delete post on guest account

### PR DESCRIPTION
**Pull request by AlisherNurlybaev**

### CODE REVIEW

File: controllers/validators/deleteCommentValidator.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const Joi = require('@hapi/joi');
+
+exports.dataShcema = Joi.object().keys({
+  id: Joi.string().required(),
+  data: Joi.object().keys({
+    root_author: Joi.string().required(),
+    permlink: Joi.string().required(),
+  }).required(),
+  userName: Joi.string(),
```
Comment (Fesmofet) [inline controllers/validators/deleteCommentValidator.js:9]: userName is not required?

File: swagger/swagger.json [hunk 2]
```diff
@@ -90,6 +90,39 @@
...
+            "required": false,
+            "schema": {
+              "example": {
+                "id":"waivio_guest_vote",
+                "data":{
+                  "root_author":"guest08",
+                  "permlink":"3blugh-sdc"
+              },
+                "userName":"waivio_alisher-nurlybae"}
+            }
```
Comment (Fesmofet) [inline swagger/swagger.json:114]: format schema

File: swagger/swagger.json [hunk 3]
```diff
@@ -90,6 +90,39 @@
...
+          }
+        ],
+        "responses" : {
+          "200" : {
+            "description" : "**SUCCESSFUL OPERATION**\n**transactionId** - id of transaction with comment which include append object\n**author** - author of recently created comment with append object\n**permlink** - permlink of recently created comment with append object\n**parentAuthor** - author of comment with create object which append related on\n**parentPermlink** - permlink of comment with create object which append related on\n",
+            "schema" : {
+              "$ref" : "#/definitions/inline_response_200_2"
```
Comment (Fesmofet) [inline swagger/swagger.json:121]: i this proper response?

File: utilities/helpers/deleteCommentHelper.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const _ = require('lodash');
+const { hiveOperations } = require('utilities/hiveApi');
+const { redisHelper } = require('utilities/redis');
+const { deleteAction } = require('constants/guestRequestsData');
+const { getAppData } = require('../waivioApi/apiRequests');
+const { operations } = require('../hiveApi/hiveOperations');
+
+exports.deleteComment = async (data) => {
+  const { app } = await getAppData('waivio');
```
Comment (Fesmofet) [inline utilities/helpers/deleteCommentHelper.js:9]: where error handle?

File: utilities/helpers/deleteCommentHelper.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  const key = _.get(_.filter(app.service_bots, (bot) => {
+    if (bot.name === data.data.root_author) return bot;
+  }), '[0].postingKey');
+
+  const operation = [['delete_comment', { author: data.data.root_author, permlink: data.data.permlink }]];
+
+  if (await hiveOperations.getAccountRC(data.data.root_author) < 2100000000) {
+    await redisHelper.addToQueueDeleteComment(data, deleteAction);
+  }
```
Comment (Fesmofet) [inline utilities/helpers/deleteCommentHelper.js:19]: you try to send operation if you know thar rc is low?

2100000000 to constants

File: utilities/helpers/deleteCommentHelper.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const { result, error } = await operations(operation, key);
```
Comment (Fesmofet) [inline utilities/helpers/deleteCommentHelper.js:20]: rename sendOperations

File: utilities/hiveApi/hiveOperations.js [hunk 7]
```diff
@@ -24,6 +24,14 @@ exports.postWithOptions = async ({ comment, options, key }) => {
   }
 };
 
+exports.operations = async (operations, key) => {
```
Comment (Fesmofet) [inline utilities/hiveApi/hiveOperations.js:27]: rename sendOperations

File: utilities/redis/redisHelper.js [hunk 8]
```diff
@@ -35,6 +35,34 @@ const addToQueue = async (data, actionData) => {
...
+  });
+  const redisDataError = await redisSetter.setActionsData(messageId, JSON.stringify(data));
+
+  if (sendMessError || redisDataError) {
+    return { error: { status: 503, message: sendMessError || redisDataError } };
+  }
+  const result = { waitingTime: await timeToPosting(actionData) };
+
+  return { result };
+};
```
Comment (Fesmofet) [inline utilities/redis/redisHelper.js:65]: DRY

File: utilities/operations/broadcastOperations.js [hunk 9]
```diff
@@ -20,17 +21,23 @@ const commentBroadcaster = async ({
...
   }
-  if (message) {
-    const result = await broadcastStatusParse(message.message, path, postingErrorWait, qname, botType);
-    if (!result) {
-      await redisQueue.deleteMessage(
-        { client: actionsRsmqClient, qname, id: message.id },
-      );
-      await redisSetter.delActionsData(message.message);
-    }
+
```
Comment (Waiviogit) [inline utilities/operations/broadcastOperations.js:24]: why remove if (message) {?

File: utilities/helpers/deleteCommentHelper.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+const _ = require('lodash');
+const { hiveOperations } = require('utilities/hiveApi');
+const { redisHelper } = require('utilities/redis');
+const { deleteAction } = require('constants/guestRequestsData');
+const { MIN_RC } = require('constants/userData');
+const { getAppData } = require('../waivioApi/apiRequests');
+const { sendOperations } = require('../hiveApi/hiveOperations');
+
+exports.deleteComment = async (data) => {
+  const { app, error } = await getAppData('waivio');
```
Comment (Waiviogit) [inline utilities/helpers/deleteCommentHelper.js:10]: why hardcode?

File: utilities/redis/redisHelper.js [hunk 11]
```diff
@@ -4,29 +4,35 @@ const redisSetter = require('utilities/redis/redisSetter');
...
+    data.author = sendData.data.root_author;
+  }
+
   const { error: createError } = await redisQueue.createQueue(
     { client: actionsRsmqClient, qname: actionData.qname },
   );
 
   if (createError) return { error: { status: 500, message: createError } };
-  const { result: currentUserComments } = await redisGetter.getHashKeysAll(`${actionData.operation}:${data.comment.author}:*`);
+  const { result: currentUserComments } = await redisGetter.getHashKeysAll(`${actionData.operation}:${data.author}:*`);
```
Comment (Waiviogit) [inline utilities/redis/redisHelper.js:21]: queueOperations.js line 9 what will you get if you call addToQueue from there?

**Review by Waiviogit — APPROVED** [review]

**Pull request merged**